server/blob/blob-s3/src/main/java/org/apache/james/blob/objectstorage/aws/S3DumbBlobStore.java

```diff
@@ -94,12 +97,21 @@
             .endpointOverride(configuration.getEndpoint())
             .region(region.asAws())
             .build();
+
+        clientPool = PoolBuilder.from(Mono.fromCallable(clientCreator))
+            .acquisitionScheduler(Schedulers.elastic())
+            .destroyHandler(client -> Mono.fromRunnable(client::close))
+            .maxPendingAcquireUnbounded()
+            .sizeBetween(4, 64)
+            .fifo();
+
+        clientPool.warmup().block();
```

Comment:
The warmup should not be done in a constructor. We should rather, if adopted, follow the Startable patern.